File: Selenium/Activities/Activity12.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.WebElement;

public class Activity12 {
    public static void main(String[] args) throws InterruptedException {
        // Setup the Firefox driver(GeckoDriver)
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        // Open a new browser and get the title of page
        driver.get("https://v1.training-support.net/selenium/dynamic-controls");
        System.out.println("Title of the page: " + driver.getTitle());
        // Find the text field and check if enabled
        WebElement textField = driver.findElement(By.id("input-text"));
        System.out.println("Is text field enabled: " +textField.isEnabled());
        // Click the "Enable Input" button
        Thread.sleep(2000);
        driver.findElement(By.id("toggleInput")).click();
        // Check if the text field is enabled
        System.out.println("Is text field enabled: " +textField.isEnabled());
        driver.quit();
    }
}

File: Selenium/Activities/Activity12.py
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the browser to the URL
    driver.get("https://v1.training-support.net/selenium/dynamic-controls")
    print("Title of the page is: ", driver.title)
    # Find the text field and check if enabled
    textField = driver.find_element(By.ID, "input-text")
    print("Is text field enabled: ", textField.is_enabled())
    # Click the "Enable Input" button
    time.sleep(2)
    driver.find_element(By.ID, "toggleInput").click()
    # Check if the text field is enabled
    print("Is text field enabled: ", textField.is_enabled())